File: python/potega.py
#!/usr/bin/env python
# -*- coding: utf-8 -*-
#
#  potega.py

def potega_it(x, n):

    i = 1
    wynik = 1
    while i <= n:
        wynik = wynik * x
        
        return wynik


# potęga _rek(a, n)= 1 dla a różnego od 0
# potęga _rek(a, n)= potega_rek(a, n - 1) * a dla n =  
def potega_rek(a, n):
    if n == 0:
        return 1
    return potega_rek(a, n - 1) * a 
 
def main(args):
    x = float(input('Podaj podstawowe potegi: '))
    n = int(input('Podaj wykładnik potęgi: '))
    assert type(n) == int
    assert silnia_it(2,2) == 2
    assert silnia_it(3,2) == 9
    assert silnia_it(4,2) == 16
    assert silnia_it(5,2) == 25
    print('Wynik:')
    print (potega_it(x, n))
    return 0

if __name__ == '__main__':
    import sys
    sys.exit(main(sys.argv))

File: cpp/potega.cpp
/*
 * potega.cpp
 * a0 = 1
 * a1 = a 
 * an = a*....*a (n - czynników) dla n zaw. N+ - {1} 
 */


#include <iostream>


using namespace std;

float potega_it(float x, int n) {
    float wynik = 1;
    for(int i = 0; i < n; i++) {
        cout << i << endl;
        wynik = wynik * x;
    }
    return wynik;    
} 

int potega_rek(int a,int n);
    if n == 0
        return 1;
    return potega_rek(a, n - 1) * a; 

int main(int argc, char **argv)
{
    float x= 0;
    int n = 0;
	//pobierz od użytkownika podstawę i wykładnik
    cout << "Podstawa: " << endl;
    cin >> x;
    cout << "Wykładnik: " << endl;
    cin >> n;
    cout << "Potęga: " << potega_it(x, n) << endl;
	return 0;
}

